moveit_plugins/moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -62,18 +67,32 @@ namespace moveit_ros_control_interface
  * @param[in] force force timeout
  * @return True if timeout duration was passed
  */
-bool checkTimeout(ros::Time& t, double timeout, bool force = false)
+bool checkTimeout(const rclcpp::Clock::SharedPtr& clock, rclcpp::Time& t, double timeout, bool force = false)
```

Comment:
This seems overly complicated and the semantics of `force` is somewhat weird here...
I would replace this function with a single-line check like:
```
static const rclcpp::Duration CONTROLLERS_TIMEOUT = rclcpp::Duration::from_seconds(1.0);

...
void discover(bool forceUpdate = false)
{
  if (forceUpdate || (node_->now() - controllers_stamp_ >= CONTROLLERS_TIMEOUT))
  {
    controllers_stamp_ = node_now();
    ...
  }
}
```